src/main/java/com/visenze/datatracking/Tracker.java

```diff
@@ -0,0 +1,178 @@
+package com.visenze.datatracking;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.Gson;
+import com.visenze.datatracking.data.DataCollection;
+import com.visenze.datatracking.data.DataTrackingResponse;
+import com.visenze.datatracking.data.DeviceData;
+import com.visenze.datatracking.data.Event;
+import com.visenze.datatracking.data.EventsBody;
+import com.visenze.datatracking.data.ResultData;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class Tracker {
+
+    public static final String TAG = "Tracker";
+    private TrackingService trackingService;
+    private String code;
+    private String uid;
+    private String sid;
+    private DeviceData mDeviceData;
+    private DataCollection dataCollection;
+
+    public Tracker(String code, String uid, String sid, boolean isCN, DataCollection dataCollection) {
```

Comment:
the uid and sid may change, how will this be passed to Tracker?

Reply:
changed the handling in sessionManager